build-image/first-boot.bash

```diff
@@ -151,26 +149,29 @@ echo "OK"
 echo -n "$(timestamp) [openHABian] Waiting for dpkg/apt to get ready... "
 if wait_for_apt_to_be_ready; then echo "OK"; else echo "FAILED"; fi
 
+firmwareBefore="$(uname -r)"
 echo -n "$(timestamp) [openHABian] Updating repositories and upgrading installed packages... "
 apt-get install --fix-broken --yes &> /dev/null
 if [[ $(eval "$(apt-get --yes upgrade &> /dev/null)") -eq 100 ]]; then
   echo -n "CONTINUING... "
   dpkg --configure --pending &> /dev/null
   apt-get install --fix-broken --yes &> /dev/null
   if apt-get upgrade --yes &> /dev/null; then
-    if is_pi; then
+    if [[ $firmwareBefore != "$(uname -r)" ]]; then
```

Comment:
will it change the output of uname -r without a reboot ?

BTW why apt-get --yes upgrade twice ?

Reply:
Yes it will, because it will fail the first time if any packages are broken so it uses `dpkg` to fix any broken packages if it does fail and then completes the upgrade fully the second time. It is essentially a safety precaution that was implemented before I joined Dev and prevents weird edge case errors so I left it in.

Reply:
> Yes it will.

Did you test that on real HW ?
Sorry for being sceptic but better safe than sorry. I had to re-build the image three times until it worked properly, and you're about to change that piece of code now.

Reply:
It should, I will double check now.

Reply:
It didn't but I found a different solution.